### PR TITLE
Media Fedora Uri Pseudo-Fields

### PR DESCRIPTION
**Pull request by dannylamb**
**GitHub Issue**: 
- Resolves https://github.com/Islandora-CLAW/CLAW/issues/881
- Resolves https://github.com/Islandora-CLAW/CLAW/issues/1109
- Follow up to https://github.com/Islandora-CLAW/islandora/pull/124

# What does this Pull Request do?

Adds the Fedora Uri pseudo-field to Media.

# What's new?
Looks up media in addition to other entity types in the GeminiLookup class.  This is done by issuing a HEAD request against the binary in Fedora to find it's `rel="describedby"` link header.

# How should this be tested?

- Pull down this PR
- Clear your cache
- You should now see the Fedora Uri pseudo-field on Media entities
Screenshot from 2019-05-02 12-20-03

@Islandora-CLAW/committers

### CODE REVIEW

File: src/GeminiLookup.php [hunk 1]
```diff
@@ -77,24 +90,72 @@ public static function create(ContainerInterface $container) {
...
+      return NULL;
+    }
+
+    $is_media = $entity->getEntityTypeId() == 'media';
+
+    // Use the entity's uuid unless it's a media,
+    // use its file's uuid instead.
+    if ($is_media) {
+      try {
+        $file = $this->mediaSource->getSourceFile($entity);
```
Comment (whikloj) [inline src/GeminiLookup.php]: Why are we linking the media using the file's UUID?

**Comment by whikloj** [comment]
@dannylamb you've got some duplication in your test setup. Might want to replace the local variables `$entity`, `$file`, `$media_source`, `$gemini_client` and `$guzzle` with private class variables so you can populate them in a function and call that from multiple locations.

**Comment by dannylamb** [comment]
The ones that never change I kept as class variables, but the ones that did I just copy/pasta'd around in all the functions.  Let me declare them all and set defaults in `setUp`. That would reduce code for sure.

**Comment by dannylamb** [comment]
Thx for keeping me on point @whikloj.  Refactored tests have been pushed up.  Let me know what you think.

**Pull request merged**